### PR TITLE
Makes borgs affected by environment pressure and heat.

### PR DESCRIPTION
**Pull request by SonixApache**
- All borgs are now affected by pressure and heat
- Default protection level is that of a hardsuit; 4053 kPa and 5000°K
- Medical, Peacekeeper, Mining and Combat borgs get double pressure protection; 8106 kPa.
- Engiborgs and emagged MoMMIs get atmos hardsuit level protection; 20,265kPa and 30000°K.
- Normal MoMMIs are protected up to infinity, which is probably not enough.
- Hud elements with warnings based off of carbon damage levels & max module protection values
- Chat warnings whenever you go over the heat / temperature limit
- Wonderful new sprites by Bilgecrank / @Kokuten

example hudcool sprites

TODO:
- [x] Make borgs take heat damage
- [x] Make borgs take pressure damage
- [x] Proper protection values
- [x] Add UI elements that work (pressure alerts, temperature alerts)
- [x] Unfuck the alert values
- [x] ~~Remove useless UI elements (O2 / Fire)~~
- [x] or make useless UI elements (~~O2~~ / Fire) functional
- [x] ~~fix borgs burning~~ (maybe in a different PR)
- [x] Make MoMMIs invulnerable to heat/pressure
- [x] Make emagged MoMMIs vulnerable
- [x] ~~Re-define infinity~~
- [x] Check if swapping cells undoes brute/**burn** damage to cell organ
- [x] Add HUD elements to MoMMIs
- [x] Improve "You're taking damage!" feedback to player
- [ ] Update wiki

🆑 
 - rscadd: Borgs are now affected by heat and pressure. Protection is based on module.
 - WIP: KEEPER MoMMIs are not affected.
 - imageadd: New borg HUD sprites by Bilgecrank / Kokuten
 - tweak: re-defined infinity
 - tweak: Borg fire indicator now actually works. Probably.

### CODE REVIEW

**Comment by PJB3005** [comment]
All MoMMIs revolting because they can't stand inside an SME anymore in 3... 2... 1...

**Comment by SonixApache** [comment]
good point, forgot about that

**Comment by jknpj** [comment]
>Make emagged MoMMIs vulnerable

HARAM

File: code/modules/mob/living/silicon/mommi/mommi_modules.dm [hunk 1]
```diff
@@ -4,6 +4,11 @@
 
 /obj/item/weapon/robot_module/mommi/New(var/mob/living/silicon/robot/R)
 	..()
+	
+	//crabs are cute and therefor exempt from burning to death
+	heat_level_max = 1.#INF
```
Comment (PJB3005) [inline code/modules/mob/living/silicon/mommi/mommi_modules.dm]: # **NO**

Comment (PJB3005) [reply code/modules/mob/living/silicon/mommi/mommi_modules.dm]: Real criticism: Use `INFINITY` and redefine `INFINITY` to `1.#INF` if you have to.

Comment (SonixApache) [reply code/modules/mob/living/silicon/mommi/mommi_modules.dm]: INNNFIINIIITTTYYYYYYYYYYYYYYYYY

Comment (Exxion) [reply code/modules/mob/living/silicon/mommi/mommi_modules.dm]: You will have to because I haven't gotten around to it yet

**Comment by jakmak6** [comment]
looks good. **MERGE WHEN**

File: code/modules/mob/living/silicon/mommi/mommi_modules.dm (hunk 1 above)
Comment (SonixApache) [reply code/modules/mob/living/silicon/mommi/mommi_modules.dm]: Okay how and where? __defines?

Comment (Exxion) [reply code/modules/mob/living/silicon/mommi/mommi_modules.dm]: Just search all files for `#define INFINITY`

**Comment by Rattlederp** [comment]
Sonix you are my favorite person, this is a great idea
Do you have all the values of at what pressure/temperature each module takes damage? ~Or should I just stop being a lazy fuck and look through the code?~
also I think this is going to get a controversial tag

**Comment by Kammerjunk** [comment]
Not too hot on it **BUT** borgs will need a serious fucking nerf if grippers go through and this is a serious nerf.
The two together feel from skimming them like they're moving carbons and sillycones closer in function - gripping for both, not so much ZAS for either. Not too hot on that either but I'll gladly defer to the people if it's what the people want. If anything, I suppose it makes plasmafires less potent for milfs

**Comment by Kammerjunk** [comment]
@Rattlederp I'm going to assume without looking through defines that ONE_ATMOSPHERE is either 101 kPa or 101.3 kPa because anything else would be weird. I might also be wrong on anything or everything on this but whatevs:
By default, they can handle pressure levels up to 5 atmospheres or about 500 kPa and the same heat levels a space suit can handle.
Engiborgs override it to atmos hardsuit level head protection and 200 atmospheres or a bit over 20000 kPa.
Medical, security, miner, syndicate (not in the game unless bus AFAIK?), and combat override pressure resistance to 40 atmospheres or just over 4000 kPa.

All in all, you'll still be a proper paramed as medborg and you'll still be able to rush into that plasmaflood for your valids as secborg but you may wanna be careful if it's on fire.

**Comment by Rattlederp** [comment]
@Kammerjunk Seems pretty reasonable, thanks

**Comment by ghost** [comment]
Skimmed but if you do this you need to put a roundstart nanopaste in every robobits in a very visible way and you need to do at least a size 36 comic sans  bold underlined italics for the steps to fix burn damage to borgs. There aren't even 10 people that can do it. I wish I could say that's a hyperbolic statement but it's not. removing the cell, screwing then using cable is too hard for most people.

Also test to make sure that swapping the cell restores burn damage on the cell now that new parts restore the lost hp. Used to be that burnt cells meant you had to nanopaste or scrap the body.

**Comment by SonixApache** [comment]
@SarahJohnson489 RTFM on the former.
Good deal on the latter, though, I'll have to look into it.

**Comment by Ephx1** [comment]
> This and Grippers just turns borgs more into wonky humans

:thinking:

**Comment by SonixApache** [comment]
@Ephx1 
Borgs are still pretty durable, much more so than humans. Should the numbers be raised?
And not being able to walk into the sun as anything but an autism crab doesn't seem so bad to me.

**Comment by Ephx1** [comment]
It's not that I'm disagreeing with this change, in fact I think it's pretty good.
It's more that the reason it exists is to enable a buff that in conjunction just removes their specialty to me.
They no longer give off the vibe of being synthetic beings, instead now being crippled humans with a lawset.

That's just me though, I'm sure plenty of others disagree.

**Comment by Chazach** [comment]
>controversial label with 12-1 for and solid discussion in comments

really silicons my metaclub

**Comment by Exxion** [comment]
Yeah it's really not that controversial
But at least [Drama] no longer exists

**Comment by ghost** [comment]
\>this has [Balance] but grippers don't

hmmmm

**Comment by SonixApache** [comment]
Hm, in testing there's no real clear indicator that you're taking damage apart from the hud indicators being maxed out. Maybe they should flash? Or a chat message?

**Comment by SonixApache** [comment]
Added warning messages. They fire once, when the indicator hits the warning level, and then stay quiet till the alarm goes away.
image

**Comment by saronsen** [comment]
What's next? Borgs tripping on ZAS? Stunned by tazers?

I mean if I wanted to play a human I would just play a human.

**Comment by SonixApache** [comment]
You play borgs to be immune to plasmafires, zas and tasers?

**Comment by saronsen** [comment]
You don't?

**Comment by SonixApache** [comment]
i play silicons to be a robutt, i dunno man

**Comment by Bilgecrank** [comment]
Borgs are still immune to cold, though and zero pressure. Ice station when.

**Comment by Uqualm** [comment]
I am glad I don't play on /vg/.

**Comment by SonixApache** [comment]
useful constructive criticism tbh

**Comment by Chazach** [comment]
@Uqualm we are glad you don't play here to

**Comment by ghost** [comment]
Death to silicons

**Comment by SonixApache** [comment]
- removed infinity change
- protection values upped; base level is hardsuit level, 'hardsuit' borgs get double that, atmos untouched

**Comment by Exxion** [comment]
Why remove the infinity change?

**Comment by SonixApache** [comment]
people told me not to do it

**Comment by gbasood** [comment]
It broke a bunch of shit so that definition of infinity was removed, yeah.

**Comment by Exxion** [comment]
I'm only aware of it breaking one thing

**Comment by gbasood** [comment]
There was at least two, one of which was naonui

**Comment by Exxion** [comment]
That's more of a problem with NanoUI, really
But I guess the define can be left like that until that gets fixed

**Comment by gbasood** [comment]
It's technically a json serialization problem
JSON doesn't have `Infinity` which is what the BYOND infinite variable serializes to naively, so when nanoui tries to load the JSON it breaks

**Comment by Exxion** [comment]
I know
Or at least I knew it was something along those lines
But we can't change how JSON or BYOND works, so the fix *we* could do would be for NanoUI

**Comment by gbasood** [comment]
We'd have to write a javascript program to parse JSON manually
that's dumb
I'd rather tell Lummox his standards-breaking code breaks JSON

**Comment by Exxion** [comment]
I'm pretty sure that the Javascript thing is neither the only nor the best way to do that
And Lummox wouldn't fix it

**Comment by SonixApache** [comment]
This is as good to go as far as I'm concerned.

**Comment by Riekland** [comment]
Wait does this mean borgs can freeze and can't go outside anymore or am I an idord

**Comment by SonixApache** [comment]
ur an idord
They're protected downwards as far as they're protected upwards. Their protection is upwards(downwards) from -5000K, and the general cap on downwards temperatures is −273C or 0K.

**Comment by Riekland** [comment]
It would be nice if borgs getting cold made them transfer that heat to peoples hands when they touch them
That might be unatomic though

**Comment by SonixApache** [comment]
outside the scope of this PR

**Comment by jakmak6** [comment]
This is still open? What's the holdup?

**Comment by SonixApache** [comment]
server poll is up

**Comment by aceedex** [comment]
Cant wait for people to start flooding every round because "well borgs are OP how else are you going to deal with them?"
Whats the point of this?

**Comment by jakmak6** [comment]
aceedex the point is to give borgs actual downsides and make sillicon blasmaflooding actually a bit more dangerous to borgs

**Comment by aceedex** [comment]
aside from having laws that can be changed at any time and not having hands and a global killswitch?
If you want  borgs to be easier to deal with maybe change the access requirement on the killswitch, have someone have the balls to make Ion pistols obtainable or what have you.
They already risk shattering any of three component that essentially kills them after 2 laser gun shots. 
All this does it put a dumb restriction on certain borg types. AI flodding is not lethal because mediborgs and standard borgs can move around in them. 

All this does its makes anyone with an atmos suit have complete free reign if they flood hard enough

**Comment by SonixApache** [comment]
unless you swap a borg to an engisuit
or unless said atmos suit man just sets fire to science and grabs an ID to blow the borgs with
non-issue, reallly

This exists because there was some code for it anyway, because it makes some sense, and to slightly nerf milf plasmaflooding.

**Comment by Kammerjunk** [comment]
>Plasmaflood filthy carbons
>Camp chemistry
>Spam the voxtest voices over comms to taunt them while they choke on it
>Only 6 suits in the game that ignore it

Wasn't it because of grippers anyway? It's dead Jim

**Comment by SonixApache** [comment]
http://ss13.moe/index.php/poll/124
Poll complete, resounding vote for NO. 

Going to strip out the damage aspects of this and throw it up with just the sensor info some time soon.

**Comment by Exxion** [comment]
I wouldn't call it a *resounding* no
But I can't really complain, because I really wasn't looking forward to fixing conflicts with XGM

**Comment by jakmak6** [comment]
j33vesu

**Pull request closed without merging**